tests/test_websockets.py

```diff
@@ -0,0 +1,151 @@
+import pytest
+import asyncio
+import websockets
+from http import HTTPStatus
+
+
+@pytest.mark.asyncio
+async def test_websocket_route(liveclient):
+    ev = asyncio.Event()
+
+    @liveclient.app.route('/ws', protocol="websocket")
+    async def handler(request, ws, **params):
+        assert ws.subprotocol is None
+        ev.set()
+
+    with liveclient as query:
+        response = await query('GET', '/ws', headers={
+            'Upgrade': 'websocket',
+            'Connection': 'upgrade',
+            'Sec-WebSocket-Key': 'hojIvDoHedBucveephosh8==',
+            'Sec-WebSocket-Version': '13'})
+
+    #assert ev.is_set()
```

Comment:
Is it set at that point?

Reply:
It is. I pushed a fix to uncomment that test. I don't remember why it was commented... and it slipped under the radar !